django_mongodb/fields.py

```diff
@@ -32,3 +49,130 @@ def to_python(self, value):
+
+
+def data_contains(self, compiler, connection):  # noqa: ARG001
+    raise NotSupportedError("contains lookup is not supported on this database backend.")
+
+
+def contained_by(self, compiler, connection):  # noqa: ARG001
+    raise NotSupportedError("contained_by lookup is not supported on this database backend.")
+
+
+def json_exact(self, compiler, connection):
+    lhs_mql = process_lhs(self, compiler, connection, bare_column_ref=True)
+    rhs_mql = json_process_rhs(self, compiler, connection)
+    if rhs_mql == "null":
+        return {"$or": [{lhs_mql: {"$eq": None}}, {lhs_mql: {"$exists": False}}]}
+    return {lhs_mql: {"$eq": rhs_mql, "$exists": True}}
+
+
+def key_transform_isnull(self, compiler, connection):
+    lhs_mql = process_lhs(self, compiler, connection, bare_column_ref=True)
+    rhs_mql = json_process_rhs(self, compiler, connection)
+
+    # https://code.djangoproject.com/ticket/32252
+    return {lhs_mql: {"$exists": not rhs_mql}}
+
+
+def key_transform_in(self, compiler, connection):
+    lhs_mql = key_transform_agg(self.lhs, compiler, connection)
+    bare_lhs_mql = process_lhs(self, compiler, connection, bare_column_ref=True)
+    value = json_process_rhs(self, compiler, connection)
+    expr = connection.mongo_aggregations[self.lookup_name](lhs_mql, value)
+    return {"$expr": expr, bare_lhs_mql: {"$exists": True}}
+
+
+def has_key_lookup(self, compiler, connection):
+    lhs = process_lhs(self, compiler, connection, bare_column_ref=True)
+    rhs = self.rhs
+    if not isinstance(rhs, (list | tuple)):
+        rhs = [rhs]
+    paths = []
+    for key in rhs:
+        if isinstance(key, KeyTransform):
+            *_, rhs_key_transforms = key.preprocess_lhs(compiler, connection)
+            rhs_key_transforms = ".".join(rhs_key_transforms)
+        else:
+            rhs_key_transforms = str(key)
+        rhs_json_path = f"{lhs}.{rhs_key_transforms}"
+        paths.append(rhs_json_path)
+
+    keys = []
+    for path in paths:
+        keys.append({path: {"$exists": True}})
+    if self.mongo_operator is None:
+        assert len(keys) == 1
+        return keys[0]
+    return {self.mongo_operator: keys}
+
+
+def key_transform_agg(self, compiler, connection):
+    key_transforms = [self.key_name]
+    previous = self.lhs
+    while isinstance(previous, KeyTransform):
+        key_transforms.insert(0, previous.key_name)
+        previous = previous.lhs
+    lhs_mql = previous.as_mql(compiler, connection)
+    result = f"{lhs_mql}"
+    for key in key_transforms:
+        get_field = {"$getField": {"input": result, "field": key}}
+        if key.isdigit():
+            result = {
+                "$cond": {
+                    "if": {"$isArray": result},
+                    "then": {"$arrayElemAt": [result, int(key)]},
+                    "else": get_field,
+                }
+            }
+        else:
+            result = get_field
+    return result
+
+
+def load_fields():
```

Comment:
I think we should keep the `register_fields()` naming convention. When I first saw the name, I wondered if it had a different from the other `register_` functions, but that doesn't seem to be the case.